docs/r2/tensorboard_quickstart.ipynb

```diff
@@ -0,0 +1,463 @@
+        "  for x_train, y_train in dataset:\n",
+        "    loss = train_step(model, optimizer, x_train, y_train)\n",
+        "    avg_loss(loss)\n",
+        "\n",
+        "    if tf.equal(optimizer.iterations % log_freq, 0):\n",
+        "      tf.summary.scalar('loss', avg_loss.result(), step=optimizer.iterations)\n",
+        "      tf.summary.scalar('accuracy', compute_accuracy.result(), step=optimizer.iterations)\n",
+        "      print('Step #%d\\tLoss: %.6f' % (optimizer.iterations, loss))\n",
+        "    avg_loss.reset_states()\n",
+        "    compute_accuracy.reset_states()\n",
+        "\n",
+        "def test(model, dataset, step_num):\n",
+        "  avg_loss = tf.keras.metrics.Mean('loss', dtype=tf.float32)\n",
+        "\n",
+        "  for (x_test, y_test) in dataset:\n",
+        "    logits = model(x_test, training=False)\n",
+        "    avg_loss(compute_loss(y_test, logits))\n",
+        "    compute_accuracy(y_test, logits)\n",
+        "  print('Model test set loss: {:0.4f} accuracy: {:0.2f}%'.format(\n",
+        "      avg_loss.result(), compute_accuracy.result() * 100))\n",
+        "  tf.summary.scalar('loss', avg_loss.result(), step=step_num)\n",
+        "  tf.summary.scalar('accuracy', compute_accuracy.result(), step=step_num)\n",
+        "    "
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "kvbPGxI2hZ4h",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Train and test the model within the scope of summary writers to write the summaries to disk."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "0wxFZp8JDXb1",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "current_time = datetime.now().strftime(\"%Y%m%d-%H%M%S\")\n",
+        "train_log_dir = 'logs/gradient_tape/' + current_time + '/train'\n",
+        "test_log_dir = 'logs/gradient_tape/' + current_time + '/test'\n",
+        "train_summary_writer = tf.summary.create_file_writer(train_log_dir)\n",
+        "test_summary_writer = tf.summary.create_file_writer(test_log_dir)\n",
+        "\n",
+        "model = create_model() # reset our model\n",
+        "\n",
+        "with train_summary_writer.as_default():\n",
+        "  train(model, optimizer, train_dataset)\n",
+        "\n",
+        "with test_summary_writer.as_default():\n",
+        "  test(model, test_dataset, optimizer.iterations)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "Fwhsw0zN23wm",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Open TensorBoard in the same way way but point it at a different log directory. After starting TensorBoard, you can rerun the previous cells to start training again and monitor the training within TensorBoard while it's happening.\n",
```

Comment:
Given that this is the second example, and also that this training
period takes noticeably longer, it might be a good time to point out
that you’ll want to start `%tensorboard` before you start training your
model lest the Colab cell lock force you to wait until training is over
entirely. (Omitting it from the very first example is totally
reasonable, IMO.)

Reply:
I tried this, but TensorBoard wasn't updating during training. Clicking refresh would show some charts but when additional epochs are completed, the chart doesn't rescale to show the additional points. Let's investigate that separately from this and update the doc later.

Reply:
> when additional epochs are completed, the chart doesn't rescale to
> show the additional points

This is true: once the window is set by the initial set of data, you
have to click the “zoom to fit” button if you want it to zoom to fit.
(That’s the third button from the left below the chart.)

I’ve gotten used to this behavior to the point where I don’t notice it,
but you’re right that it’s not actually a _good_ thing. Perhaps we can
change it. :-)